aws/table_aws_cognito_user_pool_identity_provider.go

```diff
@@ -0,0 +1,232 @@
+			return nil, err
+		}
+		for _, provider := range output.Providers {
+			plugin.Logger(ctx).Debug("aws_cognito_user_pool_identity_provider.listCognitoIdentityProviders", "provider", fmt.Sprintf("%#v", provider))
+			d.StreamListItem(ctx, provider)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getCognitoUserPoolIdentityProvider(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var providerName string
+	if h.Item != nil {
+		data := h.Item.(types.ProviderDescription)
+		if data.ProviderName != nil {
+			providerName = *data.ProviderName
+		}
+	} else {
+		providerName = d.EqualsQualString("provider_name")
+	}
+	userPoolId := d.EqualsQualString("user_pool_id")
+
+	plugin.Logger(ctx).Debug("aws_cognito_user_pool_identity_provider.getCognitoUserPoolIdentityProvider", "provider_name", providerName, "user_pool_id", userPoolId)
+
+	// check if providerName or userPoolId is empty
+	if providerName == "" || userPoolId == "" {
+		return nil, nil
+	}
+
+	// Create Session
+	svc, err := CognitoIdentityProviderClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_cognito_user_pool_identity_provider.getCognitoUserPoolIdentityProvider", "connection_error", err)
+		return nil, err
+	}
+
+	if svc == nil {
+		// Unsupported region check
+		return nil, nil
+	}
+
+	// Build the params
+	params := &cognitoidentityprovider.DescribeIdentityProviderInput{
+		ProviderName: aws.String(providerName),
+		UserPoolId:   aws.String(userPoolId),
+	}
+
+	// Get call
+	data, err := svc.DescribeIdentityProvider(ctx, params)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_cognito_user_pool_identity_provider.getCognitoUserPoolIdentityProvider", "api_error", err)
+		return nil, err
+	}
+	plugin.Logger(ctx).Debug("aws_cognito_user_pool_identity_provider.getCognitoUserPoolIdentityProvider", "identity_provider", fmt.Sprintf("%#v", *data.IdentityProvider))
+	return *data.IdentityProvider, nil
+}
+
+func getCognitoUserPoolIdentityProviderTurbotAkas(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	region := d.EqualsQualString(matrixKeyRegion)
+	userPoolId := d.EqualsQualString("user_pool_id")
+	data := h.Item.(types.ProviderDescription)
+	plugin.Logger(ctx).Debug("aws_cognito_user_pool_identity_provider.getCognitoUserPoolIdentityProviderTurbotAkas", "user_pool_id", userPoolId, "provider_name", *data.ProviderName)
+
+	// Get common columns
+
+	c, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_cognito_user_pool_identity_provider.getCognitoUserPoolIdentityProviderTurbotAkas", "common_error", err)
+		return nil, err
+	}
+	commonColumnData := c.(*awsCommonColumnData)
+
+	// Get data for turbot defined properties
+	//arn:aws:cognito-idp::<account-id>:userpool/<id>/provider/<name>
```

Comment:
Do identity providers have ARNs in AWS? I couldn't find any, but wanted to confirm that we're creating one here just for use in the AKAs.

Reply:
No, they do not have ARNs: https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_IdentityProviderType.html

Reply:
Looking at https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazoncognitouserpools.html#amazoncognitouserpools-resources-for-iam-policies, I also don't see an ARN listed for identity provider types, so makes sense to create a format that is consistent with their other ARNs